R/rowClust.R

```diff
@@ -0,0 +1,123 @@
+#' Cluster assay rows
+#'
+#' \code{rowClust} Performs clustering on assay rows of the 
+#' \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}, 
+#' by using the dissimilarity function provided at \code{dissimilarity_FUN}.
+#'
+#' @param x a
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#'   object.
+#'   
+#' @param abund_values a \code{character} vector, name of the assay containing the data.
+#' 
+#' @param name a \code{character} vector, name of the new column at \code{rowData} where to place cluster ids.
+#' 
+#' @param clustering_FUN a \code{stats::hclust} a function defined and provided by
+#'  user, to compute the clustering. Note! The function should return an \code{hclust}
+#'  object. Example functions: 'stats::hclust' or 'fastcluster::hclust'
+#' 
+#' @param dissimilarity_FUN \code{function(x) 1 - stats::cor(t(x))} a function defined and provided by
+#'  user, to compute the dissimilarity matrix. Note! the function expects an input
+#'  of type (features x samples).
+#' 
+#' @param ... other arguments passed onto \code{\link[stats:cutree]{cutree}}
+#'
+#' @return \code{x} with additional column at \code{rowData} named \code{*name*}
+#'
+#' @export
+#'
+#' @references
+#' Claesson, M., Jeffery, I., Conde, S. et al. Gut microbiota composition
+#'  correlates with diet and health in the elderly. Nature 488, 178–184 (2012)
+#' \url{https://doi.org/10.1038/nature11319}
+#'
+#' Minot, S. S., & Willis, A. D. (2019). Clustering co-abundant genes identifies
+#' components of the gut microbiome that are reproducibly associated with 
+#' colorectal cancer and inflammatory bowel disease. Microbiome, 7(1), 110.
+#' \url{https://doi.org/10.1186/s40168-019-0722-6}
+#'
+#' @name rowClust
+#' @export
+#' 
+#' @examples
+#' # Load example data
+#' library(mia)
+#' data("GlobalPatterns")
+#' tse <- GlobalPatterns
```

Comment:
This `library(mia) `might cause that error in GH action

Additionally, there are some notes so check them also